flake.nix

```
@@ -59,34 +59,48 @@
             activate.custom;
 
           activate = rec {
-            custom = base: activate: pkgs.buildEnv {
-              name = ("activatable-" + base.name);
-              paths = [
-                base
-                (pkgs.writeTextFile {
-                  name = base.name + "-activate-path";
-                  text = ''
-                    #!${pkgs.runtimeShell}
-                    set -euo pipefail
-
-                    ${activate}
-                  '';
-                  executable = true;
-                  destination = "/deploy-rs-activate";
-                })
-                (pkgs.writeTextFile {
-                  name = base.name + "-activate-rs";
-                  text = ''
-                    #!${pkgs.runtimeShell}
-                    exec ${self.defaultPackage."${system}"}/bin/activate "$@"
-                  '';
-                  executable = true;
-                  destination = "/activate-rs";
-                })
-              ];
-            };
-
-            nixos = base: custom base.config.system.build.toplevel ''
+            custom =
+              {
+                __functor = customSelf: base: activate:
+                  pkgs.buildEnv {
+                    name = ("activatable-" + base.name);
+                    paths =
+                      [
+                        base
+                        (pkgs.writeTextFile {
+                          name = base.name + "-activate-path";
+                          text = ''
+                            #!${pkgs.runtimeShell}
+                            set -euo pipefail
+
+                            if [[ $DRY_ACTIVATE == "1" ]]
```

Oh yeah, right.

@notgne2

```suggestion
                            if [[ ${DRY_ACTIVATE:-} == "1" ]]
```

---

I'm not sure `${DRY_ACTIVATE:-}` will work in the presence of `set -u`? If it does, then great. But yeah, I ran into that briefly and was losing my mind for a minute before I realized the full path of the rollback binary was local to the profile

---

Yeah, right, also needs to be quoted.